### ISSUE TITLE
[genxml] Constructor calls on fields result in '???' :value metadata

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#4088]
Tends to look strange in dox' output with `???` everywhere. :) Works correctly for static function calls.

```haxe
class Main
{
	public static var test = new Test();
	public static var test2 = Test.fromStatic();
	
	public static function main() {}
}

class Test
{
	public static function fromStatic() return new Test();
	
	public function new() {}
}
```

````xml
<test public="1" expr="&apos;???&apos;" line="4" static="1">
	<c path="Test"/>
	<meta><m n=":value"><e>'???'</e></m></meta>
</test>
<test2 public="1" expr="Test.fromStatic()" line="5" static="1">
	<c path="Test"/>
	<meta><m n=":value"><e>Test.fromStatic()</e></m></meta>
</test2>
```

**Comment by Simn** [comment]
https://github.com/HaxeFoundation/haxe/blob/development/ast.ml#L747

This is a really good opportunity to get used to OCaml!

**Comment by Gama11** [comment]
Guess the manual merge of my pull (#4098) prevented the `closes #4088` in the title from having any effect.
